typescript/src/codeActions/custom/renameParameterToNameFromType.ts

```diff
@@ -0,0 +1,87 @@
+import { pipe, groupBy, map } from 'lodash/fp'
+import { compact } from 'lodash'
+import { CodeAction } from '../getCodeActions'
+import extractType from '../../utils/extractType'
+
+const getEdits = (
+    fileName: string,
+    position: number,
+    parameterName: string,
+    parameterIndex: number,
+    functionDeclaration: ts.Node,
+    languageService: ts.LanguageService,
+): ts.FileTextChanges[] | undefined => {
+    const typeChecker = languageService.getProgram()!.getTypeChecker()
+    const type = extractType(typeChecker, functionDeclaration)
+
+    const typeSignatureParams = typeChecker.getSignaturesOfType(type, ts.SignatureKind.Call)[0]?.parameters
+    if (!typeSignatureParams) return
+
+    const typeParamName = typeSignatureParams[parameterIndex]!.name
+    if (parameterName === typeParamName) return
+
+    const renameLocations = languageService.findRenameLocations(fileName, position, false, false, {})
+    if (!renameLocations) return
+
+    const extractFileName = ({ fileName }: ts.RenameLocation) => fileName
+
+    return pipe(
+        groupBy(extractFileName),
+        Object.entries,
+        map(
+            ([fileName, changes]): ts.FileTextChanges => ({
+                fileName,
+                textChanges: changes.map(
+                    ({ textSpan }): ts.TextChange => ({
+                        newText: typeParamName,
+                        span: textSpan,
+                    }),
+                ),
+            }),
+        ),
+    )(renameLocations)
+}
+export const renameParameterToNameFromType = {
+    id: 'renameParameterToNameFromType',
+    name: 'Rename Parameter to Name from Type',
```

Comment:
How about to display preview name? Like this:

```suggestion
    name: "Rename Parameter to Name from Type 'foo'",
```

Reply:
Can you give a hint how to do this?

Reply:
Sorry, I missed this message somehow (with having a lot of other notification). I must admit I didn't do this before, however there is a tricky but IMO still clean & simple way to do this, inside of `tryToApply` function do this:

```ts
this.name = 'dynamically generated name'
```